app/models/spree/order_decorator.rb

```diff
@@ -55,8 +55,10 @@
     else
       # Find orders that are distributed by the user or have products supplied by the user
       # WARNING: This only filters orders, you'll need to filter line items separately using LineItem.managed_by
+      user_enterprise_ids = user.enterprises.select("enterprises.id").to_a
```

Comment:
why convert to array if you can keep a relation that can become a subquery? I'd just remove the .to_a here